extensions/resource-metrics/resource-metrics-extention/ui/src/Metrics/Chart/Chart.tsx

```diff
@@ -372,83 +372,149 @@ export const TimeSeriesChart = ({
 
+                      return (
+                        <ReferenceLine
+                          key={eventKey + i}
+                          isFront
+                          x={moment(eventKey).unix()}
+                          stroke="#e96d76"
+                          strokeWidth={2}
+                          strokeDasharray={"3 2"}
+                          label={
+                            <Label
+                              position="center"
+                              content={(p: any) => renderEventContent(p, event)}
+                            />
+                          }
+                        />
+                      );
+                    }
+                  )}
+                  {XAxisMemo}
+                  {YAxisMemo}
+                  {TooltipMemo}
+                  {chartData?.length > 0 ? LegendMemo : null}
+                  {nonLimitRequestData?.map((d: any, i: number) => {
                     return (
-                      <ReferenceLine
-                        key={eventKey + i}
-                        isFront
-                        x={moment(eventKey).unix()}
-                        stroke="#e96d76"
-                        strokeWidth={2}
-                        strokeDasharray={"3 2"}
-                        label={
-                          <Label
-                            position="center"
-                            content={(p: any) => renderEventContent(p, event)}
-                          />
+                      <Line
+                        // strokeDasharray={`${strokeArray(i)}`}
+                        isAnimationActive={false}
+                        dataKey="y"
+                        data={d.data}
+                        connectNulls={false}
+                        hide={
+                          filterChart[groupBy] &&
+                          filterChart[groupBy].indexOf(d.name) < 0
                         }
+                        stroke={colorArray[i % colorArray.length]}
+                        strokeWidth={d.name === highlight[groupBy] ? 3 : 1.5}
+                        name={d.name}
+                        dot={false}
+                        key={d.name}
+                        animationDuration={200}
+                        style={{ zIndex: highlight[groupBy] ? 100 : 1 }}
                       />
                     );
+                  })}
+                </LineChart>
+              </ResponsiveContainer>
+              <ul
+                style={{
+                  display: "flex",
+                  fontSize: "12px",
+                  justifyContent: "center",
+                  alignItems: "center",
+                  marginTop: "10px",
+                  listStyle:'none',
+                }}
+              >
+                {limitRequestData?.map((d: any, i: number) => {
+                  let metricName = "";
+                  let metricValue = "";
+
+                  if (d.metric.__name__.endsWith("requests")) {
```

Comment:
This code shouldn't be based on hardcoded strings verifying patterns in the metric name. As this is an open-source app, we can't predict how metrics will be named. Instead, this logic to show request limits and specific legends should be based on the graph configuration defined by the user. This is one example of how the configuration looks like:
https://github.com/argoproj-labs/argocd-extension-metrics/blob/main/manifests/configmap.yaml#L19-L50
This configuration is returned by the `/dashboards` API to the UI extension.

Reply:
Built a completely different logic th handle Req and Limits